### PR TITLE
Upgrade to TypeScript 5.7

### LINKED ISSUE #148: TS2687 and TS2717 errors in TypeScript 4.7 with ES2024
\@zenfs/core 1.4.2 is incompatible with TypeScript 4.7 when target and lib are set to "ES2024".
TypeScript compiler raises TS2687 and TS2717 errors.

## package.json

```json
{
  "private": true,
  "type": "module",
  "packageManager": "pnpm@9.14.2+sha512.6e2baf77d06b9362294152c851c4f278ede37ab1eba3a55fda317a4a17b209f4dbb973fb250a77abc463a341fcb1f17f17cfa24091c4eb319cda0d9b84278387",
  "dependencies": {
    "@types/node": "20.16.11",
    "@zenfs/core": "^1.4.2",
    "typescript": "~5.7.2"
  }
}
```

## tsconfig.json

```jsonc
{
  "compilerOptions": {
    "forceConsistentCasingInFileNames": true,
    "lib": [
      "ES2024",
      "ESNext.Decorators",
      "ESNext.Promise"
    ],
    "module": "ES2022",
    "moduleResolution": "Bundler",
    "noEmitOnError": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "strict": true,
    "target": "ES2024",
    "types": [
      "node"
    ]
  },
  "include": [
    "main.ts"
  ],
  "exclude": [
    "**/node_modules"
  ]
}
```

The errors do not occur if "ES2024" is changed to "ES2023".

## main.ts

```ts
import { fs } from "@zenfs/core";

console.log(typeof fs.read);
```

## Command Line

```bash
./node_modules/.bin/tsc --noEmit
```

## Error Message

```text
node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:12:18 - error TS2687: All declarations of 'resizable' must have identical modifiers.

12         readonly resizable: boolean;
                    ~~~~~~~~~

node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:13:18 - error TS2687: All declarations of 'maxByteLength' must have identical modifiers.

13         readonly maxByteLength?: number;
                    ~~~~~~~~~~~~~

node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:13:18 - error TS2717: Subsequent property declarations must have the same type.  Property 'maxByteLength' must be of type 'number', but here has type 'number | undefined'.

13         readonly maxByteLength?: number;
                    ~~~~~~~~~~~~~

  node_modules/.pnpm/typescript@5.7.2/node_modules/typescript/lib/lib.es2024.arraybuffer.d.ts:25:9
    25     get maxByteLength(): number;
               ~~~~~~~~~~~~~
    'maxByteLength' was also declared here.

node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:18:18 - error TS2687: All declarations of 'maxByteLength' must have identical modifiers.

18         readonly maxByteLength?: number;
                    ~~~~~~~~~~~~~

node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:18:18 - error TS2717: Subsequent property declarations must have the same type.  Property 'maxByteLength' must be of type 'number', but here has type 'number | undefined'.

18         readonly maxByteLength?: number;
                    ~~~~~~~~~~~~~

  node_modules/.pnpm/typescript@5.7.2/node_modules/typescript/lib/lib.es2024.sharedmemory.d.ts:56:9
    56     get maxByteLength(): number;
               ~~~~~~~~~~~~~
    'maxByteLength' was also declared here.


Found 5 errors in the same file, starting at: node_modules/.pnpm/@zenfs+core@1.4.2/node_modules/@zenfs/core/dist/file.d.ts:12
```

The error message suggests that the ambient declarations in `file.d.ts` are different from TypeScript's own declarations.

### PR DESCRIPTION
**Pull request by yoursunny**
Delete ArrayBuffer and SharedArrayBuffer ambient declarations.

fixes #148

### CODE REVIEW

**Comment by yoursunny** [comment]
Needs: https://github.com/james-pre/utilium/pull/3

File: package.json [hunk 1]
```diff
@@ -72,7 +72,7 @@
 		"buffer": "^6.0.3",
 		"eventemitter3": "^5.0.1",
 		"readable-stream": "^4.5.2",
-		"utilium": "^1.0.0"
+		"utilium": "file:../utilium"
```
Comment (james-pre) [inline package.json:75]: This needs to be fixed.

Comment (yoursunny) [reply package.json]: Changed to `^1.1.1` that includes the required declaration.

File: src/file.ts [hunk 2]
```diff
@@ -449,8 +433,10 @@ export class PreloadFile<FS extends FileSystem> extends File<FS> {
 		if (end > this.stats.size) {
 			this.stats.size = end;
 			if (end > this._buffer.byteLength) {
-				if (this._buffer.buffer.resizable && this._buffer.buffer.maxByteLength! <= end) {
-					this._buffer.buffer.resize(end);
+				if ((this._buffer.buffer as ArrayBuffer).resizable && this._buffer.buffer.maxByteLength! <= end) {
```
Comment (james-pre) [inline src/file.ts:436]: If possible, I'd like to avoid the casting here, and perform a type-safe check instead.

File: src/file.ts [hunk 3]
```diff
@@ -12,22 +12,6 @@ import { _chown, Stats } from './stats.js';
...
-	interface SharedArrayBuffer {
-		readonly resizable: boolean;
-
-		readonly maxByteLength?: number;
-
-		resize(newLength: number): void;
-	}
-
 	interface ArrayBufferConstructor {
 		new (byteLength: number, options: { maxByteLength?: number }): ArrayBuffer;
```
Comment (james-pre) [inline src/file.ts:16]: This should be able to be removed as well.

Comment (james-pre) [reply src/file.ts]: Fixed in [`9461e61` (#149)](https://github.com/zen-fs/core/pull/149/commits/9461e61272f476c6426007aa3d1efccf46884c15)

**Review by james-pre — APPROVED** [review]
Okay, everything looks good now.

**Pull request merged**